### PR TITLE
[SAM] higanbana

### PR DESCRIPTION
**Pull request by Kagekazu**
- [x] Add Higanbana boss check
- [x] Add Nullcheck Target

### CODE REVIEW

File: WrathCombo/Combos/PvE/SAM/SAM_Helper.cs [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+#endregion
+
 namespace WrathCombo.Combos.PvE;
 
 internal partial class SAM
 {
     internal static SAMGauge gauge = GetJobGauge<SAMGauge>();
     internal static SAMOpenerLogic SAMOpener = new();
+    internal static bool TargetIsBoss = IsBoss(LocalPlayer.TargetObject);
+    internal static bool hasHostileTarget = HasTarget() && CurrentTarget.IsHostile();
```
Comment (Taurenkey) [inline WrathCombo/Combos/PvE/SAM/SAM_Helper.cs:22]: Might be best if these sort of things are taken out of class specific helpers and put into the custom combo stuff as overrides.

Comment (Kagekazu) [reply WrathCombo/Combos/PvE/SAM/SAM_Helper.cs:22]: Oh ye, I'll move em after dinner, then everyone can just use em. Smart cookie 😁

File: WrathCombo/CustomCombo/Functions/Target.cs [hunk 2]
```diff
@@ -571,8 +571,12 @@ internal unsafe static bool IsInLineOfSight(IGameObject target)
...
 
-        internal unsafe static bool IsQuestMob(IGameObject target) => target.Struct()->NamePlateIconId is 71204 or 71144 or 71224 or 71344;
+        internal static unsafe bool IsQuestMob(IGameObject target) => target.Struct()->NamePlateIconId is 71204 or 71144 or 71224 or 71344;
 
-        internal unsafe static bool IsBoss(IGameObject target) => Svc.Data.GetExcelSheet<BNpcBase>()?.GetRow(target.DataId).Rank is 2 or 6;
+        internal static bool IsBoss(IGameObject? target) => Svc.Data.GetExcelSheet<BNpcBase>().GetRow(target.DataId).Rank is 2 or 6;
+        
+        internal static bool TargetIsBoss = IsBoss(LocalPlayer.TargetObject);
+
+        internal static bool hasHostileTarget = HasTarget() && CurrentTarget.IsHostile();
```
Comment (zbee) [inline WrathCombo/CustomCombo/Functions/Target.cs:580]: Standard here would be for this to be named `TargetIsHostile`

File: WrathCombo/CustomCombo/Functions/Target.cs [hunk 3]
```diff
@@ -571,8 +571,12 @@ internal unsafe static bool IsInLineOfSight(IGameObject target)
             return isLoSBlocked == false;
         }
 
-        internal unsafe static bool IsQuestMob(IGameObject target) => target.Struct()->NamePlateIconId is 71204 or 71144 or 71224 or 71344;
+        internal static unsafe bool IsQuestMob(IGameObject target) => target.Struct()->NamePlateIconId is 71204 or 71144 or 71224 or 71344;
 
-        internal unsafe static bool IsBoss(IGameObject target) => Svc.Data.GetExcelSheet<BNpcBase>()?.GetRow(target.DataId).Rank is 2 or 6;
+        internal static bool IsBoss(IGameObject? target) => Svc.Data.GetExcelSheet<BNpcBase>().GetRow(target.DataId).Rank is 2 or 6;
+        
+        internal static bool TargetIsBoss = IsBoss(LocalPlayer.TargetObject);
```
Comment (zbee) [inline WrathCombo/CustomCombo/Functions/Target.cs:578]: I think this ought to be the same as `hasHostileTarget` below, where it also checks if there is a target. i.e. `HasTarget() && IsBoss(CurrentTarget)`

File: WrathCombo/CustomCombo/Functions/Target.cs [hunk 4]
```diff
@@ -571,8 +571,12 @@ internal unsafe static bool IsInLineOfSight(IGameObject target)
             return isLoSBlocked == false;
         }
 
-        internal unsafe static bool IsQuestMob(IGameObject target) => target.Struct()->NamePlateIconId is 71204 or 71144 or 71224 or 71344;
+        internal static unsafe bool IsQuestMob(IGameObject target) => target.Struct()->NamePlateIconId is 71204 or 71144 or 71224 or 71344;
 
-        internal unsafe static bool IsBoss(IGameObject target) => Svc.Data.GetExcelSheet<BNpcBase>()?.GetRow(target.DataId).Rank is 2 or 6;
+        internal static bool IsBoss(IGameObject? target) => Svc.Data.GetExcelSheet<BNpcBase>().GetRow(target.DataId).Rank is 2 or 6;
```
Comment (zbee) [inline WrathCombo/CustomCombo/Functions/Target.cs:576]: If we're adding the property `TargetIsBoss`, I think this method ought to be private.

**Review by zbee — CHANGES_REQUESTED** [review]
Requested changes on the Target methods, the rest I only gave a cursory look over but it LGTM.

File: WrathCombo/CustomCombo/Functions/Target.cs (hunk 2 above)
Comment (Kagekazu) [reply WrathCombo/CustomCombo/Functions/Target.cs:580]: Renamed.

File: WrathCombo/CustomCombo/Functions/Target.cs (hunk 3 above)
Comment (Kagekazu) [reply WrathCombo/CustomCombo/Functions/Target.cs]: Added them

File: WrathCombo/CustomCombo/Functions/Target.cs (hunk 2 above)
Comment (Kagekazu) [reply WrathCombo/CustomCombo/Functions/Target.cs]: fixed

**Review by zbee — APPROVED** [review]

**Pull request merged**